### DISCUSSION TITLE
Parallel Route in App Router

### DISCUSSION THREAD

**Discussion by sinhapaurush** [vercel/next.js#54191] [App Router] [answered]
I am learning Next JS's Parallel Route, I got stuck into it. I was testing it without using `src` directory, Parallel Router acted weird without `src` directory, but when I created new project with `src` directory it worked, I want to know why? Do I need some additional configuration for parallel routes to use it without `src` directory or is it some weird bug untested on same scenario?

**Comment by sinhapaurush** [comment, marked as answer]
I got the solution, just delete `.next` folder and restart development server.

**Comment by duoluodexiaoxiaoyuan** [comment]
when i write some content about router in app/layout.js .   Page always  alert 404 not found. I try to any methods. I don't slove it now.

**Reply by sinhapaurush** [reply]
Are you sure you have created `page.js` for respective path?

**Reply by sinhapaurush** [reply]
Sharing code would help

**Reply by duoluodexiaoxiaoyuan** [reply]
Thank your reply, this is my repo: https://github.com/duoluodexiaoxiaoyuan/mynextproject
![image](https://github.com/vercel/next.js/assets/59086082/c42f62e8-ba6f-4627-8476-e608b30d87c9)app/layout.js
```javascript
import './globals.css'
import { Inter } from 'next/font/google'
import { Suspense } from 'react'
import Loading from '@/app/loading.js'

const inter = Inter({ subsets: ['latin'] })

export const metadata = {
  title: 'Create Next App',
  description: 'Generated by create next app',
}
// 默认写的是{children}，但是我们写成props的时候要把{}去掉
export default function RootLayout(props) {
  // console.log(JSON.stringify(props)+'rootlayout')
  return (
    <html lang="en">
      <body className={inter.className}>
        nihaoshijie
        {props.children}
        {/* {console.log('独一无二')} */}
        {props.modal}
        {/* {props.team}
        {props.analytics} */}
        {/* <Suspense fallback={<Loading/>}>
          {props.children}
          {props.team}
          {props.analytics}
        </Suspense> */}
      </body>
    </html>
  )
}

```

**Reply by sinhapaurush** [reply]
Check this repo: https://github.com/sinhapaurush/mynextproject I have forked your repo and solved issues, check commit changes that I have made in this repo. I am sending you pull request for it.

**Reply by sinhapaurush** [reply]
Check this docs: https://nextjs.org/docs/app/building-your-application/routing/parallel-routes
Also if you need any help feel free to ask me!

**Comment by duoluodexiaoxiaoyuan** [comment]
> Check this docs: https://nextjs.org/docs/app/building-your-application/routing/parallel-routes Also if you need any help feel free to ask me!

very thank u
